File: CardMasterSkin/GraphicsElements/ImageElement.vb
Imports System.Drawing
Imports System.Drawing.Drawing2D
Imports CardMasterSkin.Skins

Namespace GraphicsElement

    Public Class ImageElement
        Inherits GraphicElement

        Private m_image As Image
        Private m_rectangle As Rectangle

        Public Sub New(image As Image, rectangle As Rectangle)
            MyBase.New

            m_image = image
            m_rectangle = rectangle

        End Sub

        Public Overrides Sub Draw(g As Graphics)

            g.DrawImage(m_image, m_rectangle)

        End Sub

        Public Overrides Sub DrawShadow(g As Graphics, shadow As SkinShadow)

        End Sub
    End Class

End Namespace

File: CardMasterSkin/GraphicsElements/ImageElement.cs
using System.Drawing;
using CardMasterSkin.Skins;

namespace CardMasterSkin.GraphicsElements
{
    public class ImageElement : GraphicElement
    {
        private Image image;
        private Rectangle rectangle;

        public ImageElement(Image image, Rectangle rectangle)
        {
            this.image = image;
            this.rectangle = rectangle;
        }

        ~ImageElement()
        {
            this.image = null;
        }

        public override void Draw(Graphics g)
        {
            g.DrawImage(this.image, this.rectangle);
        }

        public override void DrawShadow(Graphics g, SkinShadow shadow)
        { }

    }
}
